src/test/java/net/snowflake/client/jdbc/StatementLatestIT.java

```diff
@@ -296,4 +297,42 @@ public void testQueryIdIsSetOnFailedExecuteQuery() throws SQLException {
       }
     }
   }
+
+  /**
+   * Test for setting query timeout on async queries. Applicable to versions after 3.20.0.
+   *
+   * @throws SQLException if there is an error when executing
+   */
+  @Test
+  public void testSetQueryTimeoutForAsnycQuery() throws SQLException {
+    try (Statement statement = connection.createStatement()) {
+      statement.unwrap(SnowflakeStatement.class).setParameter("MULTI_STATEMENT_COUNT", 0);
+      statement.setQueryTimeout(3);
+
+      String sql = "SELECT * FROM SNOWFLAKE_SAMPLE_DATA.TPCDS_SF100TCL.CUSTOMER;";
+
+      try (ResultSet resultSet =
+          statement.unwrap(SnowflakeStatement.class).executeAsyncQuery(sql)) {
+
+        QueryStatus queryStatus = QueryStatus.RUNNING;
+        while (queryStatus == QueryStatus.RUNNING
+            || queryStatus == QueryStatus.RESUMING_WAREHOUSE) {
```

Comment:
resuming warehouse it's not an error we should check - if we want to avoid it let's run a normal query without timeout before to ensure that Warehouse is ready